Remove blaze link on posts and pages

CodeyGuyDylan opened the pull request:
## Proposed changes:

* Remove 'Promote with Blaze' from page and post items on the Pages page and Posts page

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

pbNhbs-9Bz-p2

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

1. Checkout this branch via the Jetpack Beta plugin or your local dev environment
2. Make sure Blaze is enabled on your site (`wp-admin/admin.php?page=jetpack#/traffic`)
![image](https://github.com/Automattic/jetpack/assets/65001528/31d1fee8-8755-4f88-be2f-d1b9715a20ce)
3. Go to `/wp-admin/edit.php` and hover over the a post. Make sure "Promote with Blaze" does not show up
![image](https://github.com/Automattic/jetpack/assets/65001528/06e0b5d7-32e8-40e7-99b1-1d8030170b94)
4. Go to `/wp-admin/edit.php?post_type=page` and make sure it doesn't show up there either
![image](https://github.com/Automattic/jetpack/assets/65001528/48b8ce54-9ec8-40ef-a3ac-5f3f209a93a9)

CodeyGuyDylan committed:
Remove blaze link from post and page items

CodeyGuyDylan committed:
changelog

CodeyGuyDylan committed:
update project version

CodeyGuyDylan committed:
Remove test that ensures blaze link is added

jeherve reviewed (commented):
I am thinking we could leave the feature in the package, so the quick action links could be added by other plugins using the package. What do you think about gating the links behind a new filter, and use the filter in `projects/plugins/jetpack/modules/blaze.php` to disable the links?

elliottprogrammer reviewed (dismissed):
Confirmed the "Promote with Blaze" links are gone. Code removal looks all good to me.
LGTM! 👍

jeherve commented:
> I am thinking we could leave the feature in the package, so the quick action links could be added by other plugins using the package. What do you think about gating the links behind a new filter, and use the filter in `projects/plugins/jetpack/modules/blaze.php` to disable the links?

I opened #35620 to give you a better idea of what I had in mind.

CodeyGuyDylan committed:
Merge branch 'trunk' into remove/blaze-link-on-posts-and-pages

jeherve committed:
Blaze: hide Post Links in the Jetpack plugin while still leaving them in other environments (#35620)

* Restore quick action post links, and introduce filter instead

Instead of removing the links entirely (which may be seen as a breaking change for other plugins and sites using the feature), let's leave the feature in, but introduce a new filter that can be used to disable the links, like so:

`add_filter( 'jetpack_blaze_post_row_actions_enable', '__return_false' );`

* Start using the filter in the Jetpack plugin

* Bump version

jeherve committed:
Merge remote-tracking branch 'origin/trunk' into remove/blaze-link-on-posts-and-pages
